[TS] Content: Add support for Firestorm Assets and Music

LipkeGu opened the pull request:
This PR is basically a PR of #11787, #12047, #12048 in one PR to fix the mirror problems
and enables the Installation and Download of "stripped down" Firestorm Assets and Music.

Most work and thanks goes to @Mailaender

pchote commented on mods/ts/installer/downloads.yaml:
```diff
@@ -1,4 +1,4 @@
-basefiles: Base Freeware Content
+tibsun: C&C Tiberian Sun
```
Please keep the displayed names as they are.  The naming (a) makes it clear that the original games were released as freeware, and (b) keeps consistency between the mods.

LipkeGu replied:
updated

pchote commented on mods/ts/installer/downloads.yaml:
```diff
@@ -15,7 +15,48 @@ basefiles: Base Freeware Content
+		^Content/ts/firestorm/e01vox01.mix: firestorm/e01vox01.mix
+		^Content/ts/firestorm/e01vox02.mix: firestorm/e01vox02.mix
+		^Content/ts/firestorm/ecache01.mix: firestorm/ecache01.mix
+		^Content/ts/firestorm/isotemp.mix: firestorm/isotemp.mix
```
Please extract out just the unique files from inside these mixes too.

LipkeGu replied:
updated

LipkeGu commented:
![fona](https://cloud.githubusercontent.com/assets/2057932/18848009/bcc5f0c6-842c-11e6-8d10-e6e36669697a.png)

LipkeGu commented:
updated

LipkeGu commented:
Can anyone share the tibsun.mix and expand01.mix from orign-installations?

LipkeGu commented:
@Mailaender I gave you Write Access to this Branch so you can work with this entire Branch... Reaper or Jugglernaut.

Mailaender commented:
Thanks for taking this task over and for the invitation, but I won't just interfere here.

LipkeGu commented:
updated

pchote commented on mods/ts/installer/origin.yaml:
```diff
@@ -50,4 +50,4 @@ origin: C&C The Ultimate Collection (Origin version, English)
 			^Content/ts/temperat.mix:
 				Offset: 73056940
-				Length: 2037606
+				Length: 2037606
```
You've accidentally removed the end-of-file newline here.

pchote commented on mods/ts/installer/tibsun.yaml:
```diff
@@ -98,3 +98,4 @@ tibsun-linux: Tiberian Sun (GDI or Nod Disc, English)
 			^Content/ts/temperat.mix:
 				Offset: 73056940
 				Length: 2037606
+
```
This change can be dropped too.

reaperrr replied:
@LipkeGu You haven't adressed this yet.

pchote commented on mods/ts/installer/downloads.yaml:
```diff
@@ -15,7 +15,202 @@ basefiles: Base Freeware Content
 		^Content/ts/scores.mix: scores.mix
+
+fstorm: Expansion Freeware Content
+	URL: http://81.169.246.181/openra/ts/ts-expand.zip
```
Can you please add these to https://github.com/OpenRA/OpenRAWeb/pull/316?  You'll need to create a new mirror txt file. Once that is merged we can set these properly.

pchote reviewed (commented):
This all looks correct, but I haven't verified whether this includes _all_ the unique firestorm content.  @ABrandau can you double check this?

Please see the nits below.

pchote commented:
:+1: after the nits have been fixed (assuming no other nits work their way in as part of that).

Commit:
Add the Metadata for the Installer

Commit:
Add the Firestorm Trees

Commit:
Add the Firestorm Tileset

Commit:
Add Map "Drawbridges"

LipkeGu commented:
Updated and switched to MirrorList for the time until the mirrors will be updated the links are stored here

firestorm files: http://81.169.246.181/openra/ts/ts-expand.zip
firestorm Music: http://81.169.246.181/openra/ts/ts-fsmusic.zip

ABrandau commented:
Dawnbridges loads fine but the quick install function didn't work for me, seems it downloaded some files and then discarded them, created a loop, I had to use the advanced install option to get the Firestorm data.

Mailaender commented:
> Most work and thanks goes to @Mailaender

Actually most work and thanks goes to @ABrandau: https://github.com/ABrandau/OpenRA/tree/FS-Files-Inclussion

> Updated and switched to MirrorList for the time until the mirrors will be updated

https://github.com/OpenRA/OpenRAWeb/pull/316 has been merged. Hope this helps.

Iamgoofball commented:
I've got the Origin collection US edition if you still need those mixes

Mailaender commented:
The purpose of this pull request is not to generate those mix files, but find a procedure so anyone can do it himself with a simple graphical user interface. You can help test this patch and give your :+1: if it works for you.

LipkeGu commented:
>Dawnbridges loads fine but the quick install function didn't work for me, seems it downloaded some files >and then discarded them, created a loop, I had to use the advanced install option to get the Firestorm >data.

it is currently not validated that the orign US mixes are valid, we need hashes can you send me the sha1 hash of expand01 or the file expand01 file itself?

the files gets extracted not copied

reaperrr commented on mods/ts/sequences/trees.yaml:
```diff
@@ -30,6 +30,17 @@ bigblue:
+bigblue3:
+	Defaults:
+		Offset: 0, -12, 12
+	idle: bigblue2
```
Why not use bigblue3?

pchote commented on mods/ts/mod.yaml:
```diff
@@ -266,19 +267,29 @@ ModContent:
 	QuickDownload: basefiles
```
You need to package a new quickinstall zip that contains all the essential content (base + firestorm, without music), and then reference it here.

pchote replied:
This is why the quick install button doesn't work.

pchote commented:
Any progress here @LipkeGu?

docholiday commented:
Hi there, the firestorm files and music seem to be down (I always get a timeout). Could you please make them available again?

reaperrr commented:
Superseded by #12429.
